pkg/codegen/java/gen_program_expressions.go

```diff
@@ -83,7 +83,27 @@ func (g *generator) GenAnonymousFunctionExpression(w io.Writer, expr *model.Anon
 	case 0:
 		g.Fgen(w, "()")
 	case 1:
-		g.Fgenf(w, "%s", expr.Signature.Parameters[0].Name)
+		// rename the parameter to avoid conflicts existing variables
+		// for example we could have data.applyValue(data -> data.result())
+		// in this case we rename the parameter to _data
+		// it becomes data.applyValue(_data -> _data.result())
+		paramName := expr.Signature.Parameters[0].Name
+		modifiedParamName := "_" + paramName
```

Comment:
I suspect this isn't safe in general (there might be another variable named `_data` in scope, for instance) but it's probably a good starting point for now. Probably should add another conformance test around shadowing etc. in future.